lib/src/util/firestore_encoding.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:firedart/firedart.dart';
+import 'package:firedart/src/generated/google/firestore/v1/document.pb.dart'
+    as fs;
+import 'package:firedart/src/generated/google/protobuf/struct.pbenum.dart';
+import 'package:firedart/src/generated/google/protobuf/timestamp.pb.dart';
+import 'package:firedart/src/repository/firestore_gateway.dart';
+import 'package:fixnum/fixnum.dart';
+
+/// An abstract util class which provides firestore decode and encode static methods.
+abstract class FirestoreEncoding {
+  /// Encode dart object into related [fs.Value].
+  static fs.Value encode(dynamic value) {
+    if (value == null) return fs.Value()..nullValue = NullValue.NULL_VALUE;
+
+    var type = value.runtimeType;
+
+    if (type is List) {
+      var array = fs.ArrayValue();
+      (value as List).forEach((element) => array.values.add(encode(element)));
+      return fs.Value()..arrayValue = array;
+    }
```

Comment:
```suggestion
    if (value is List) {
      var array = fs.ArrayValue();
      value.forEach((element) => array.values.add(encode(element)));
      return fs.Value()..arrayValue = array;
    }
```
Maybe this would be better for static and runtime analysis? The same for the statements below